config/config_mgmt.py

```diff
@@ -194,8 +197,11 @@ def readConfigDB(self):
         self.sysLog(doPrint=True, msg='Reading data from Redis configDb')
         # Read from config DB on sonic switch
         data = dict()
-        configdb = ConfigDBConnector()
-        configdb.connect()
+        if self.configdb is None:
+            configdb = ConfigDBConnector()
```

Comment:
>configdb = ConfigDBConnector() [](http://example.com/codeflow?start=12&length=30)

If initializing the `self.configdb` in `__init__()`, there is no need to initialize in other functions.

Reply:
That's true. But I remove the initialization in `__init__()` because there are cases that configdb is not required to be initialized. In that case, the ConfigDBConnector will fail to initialize if we keep that in `__init__`. Thus it will report failure.

The previous PR is blocking submodule update due to this line: https://github.com/sonic-net/sonic-utilities/blob/master/sonic_package_manager/manager.py#L1009

The error log saying Sonic database config file doesn't exist.
Please check below:
```
2023-05-01T20:07:22.5949836Z + sudo LANG=C DOCKER_HOST= chroot ./fsroot-vs /usr/local/bin/generate_shutdown_order.py
2023-05-01T20:07:22.5949999Z Traceback (most recent call last):
2023-05-01T20:07:22.5950254Z   File "/usr/local/lib/python3.9/dist-packages/config/config_mgmt.py", line 60, in __init__
2023-05-01T20:07:22.5950411Z     self.configdb = ConfigDBConnector()
2023-05-01T20:07:22.5950664Z   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 2226, in __init__
2023-05-01T20:07:22.5950872Z     super(ConfigDBConnector, self).__init__(use_unix_socket_path = use_unix_socket_path, namespace = namespace)
2023-05-01T20:07:22.5951169Z   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1855, in __init__
2023-05-01T20:07:22.5951322Z     for db_name in self.get_db_list():
2023-05-01T20:07:22.5951582Z   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 1791, in get_db_list
2023-05-01T20:07:22.5951757Z     return _swsscommon.SonicV2Connector_Native_get_db_list(self)
2023-05-01T20:07:22.5952042Z RuntimeError: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json
2023-05-01T20:07:22.5952138Z 
2023-05-01T20:07:22.5952265Z During handling of the above exception, another exception occurred:
2023-05-01T20:07:22.5952449Z Traceback (most recent call last):
2023-05-01T20:07:22.5952593Z   File "/usr/local/bin/generate_shutdown_order.py", line 15, in <module>
2023-05-01T20:07:22.5952729Z     main()
2023-05-01T20:07:22.5952854Z   File "/usr/local/bin/generate_shutdown_order.py", line 8, in main
2023-05-01T20:07:22.5953011Z     manager = PackageManager.get_manager()
2023-05-01T20:07:22.5953286Z   File "/usr/local/lib/python3.9/dist-packages/sonic_package_manager/manager.py", line 1009, in get_manager
2023-05-01T20:07:22.5953516Z     cfg_mgmt = config_mgmt.ConfigMgmt(source=INIT_CFG_JSON, sonicYangOptions=ly.LY_CTX_DISABLE_SEARCHDIR_CWD)
2023-05-01T20:07:22.5953809Z   File "/usr/local/lib/python3.9/dist-packages/config/config_mgmt.py", line 72, in __init__
2023-05-01T20:07:22.5953997Z     self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
2023-05-01T20:07:22.5954265Z   File "/usr/local/lib/python3.9/dist-packages/config/config_mgmt.py", line 158, in sysLog
2023-05-01T20:07:22.5954443Z     if self.DEBUG == False and logLevel == syslog.LOG_DEBUG:
2023-05-01T20:07:22.5954670Z AttributeError: 'ConfigMgmt' object has no attribute 'DEBUG'
```

Reply:
Offline discussed. Master branch need further refactoring.

Reply:
> Offline discussed. Master branch need further refactoring.

Hi @qiluo-msft , I prefer to keep the current structure. 
In writeDB, we use `connect(false)` as we don't need to wait for db init. 
In readDB, we need to wait for that. 
The current code is more efficient this way. 
```
def connect(self, wait_for_init=True, retry_on=False):
        self.db_connect('CONFIG_DB', wait_for_init, retry_on)
```